server/src/main/java/org/opensearch/action/admin/cluster/remotestore/RemoteStoreNode.java

```diff
@@ -0,0 +1,128 @@
+    public static final String REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY = "remote_store.translog.repository";
+    public static final String REMOTE_STORE_REPOSITORY_TYPE_ATTRIBUTE_KEY_FORMAT = "remote_store.repository.%s.type";
+    public static final String REMOTE_STORE_REPOSITORY_SETTINGS_ATTRIBUTE_KEY_PREFIX = "remote_store.repository.%s.settings.";
+    private final RepositoriesMetadata repositoriesMetadata;
+    private final DiscoveryNode node;
+
+    /**
+     * Creates a new {@link RemoteStoreNode}
+     */
+    public RemoteStoreNode(DiscoveryNode node) {
+        this.node = node;
+        this.repositoriesMetadata = buildRepositoriesMetadata();
+    }
+
+    private String validateAttributeNonNull(String attributeKey) {
+        String attributeValue = node.getAttributes().get(attributeKey);
+        if (attributeValue == null || attributeValue.isEmpty()) {
+            throw new IllegalStateException("joining node [" + this.node + "] doesn't have the node attribute [" + attributeKey + "]");
+        }
+
+        return attributeValue;
+    }
+
+    private Map<String, String> validateSettingsAttributesNonNull(String repositoryName) {
+        String settingsAttributeKeyPrefix = String.format(
+            Locale.getDefault(),
+            REMOTE_STORE_REPOSITORY_SETTINGS_ATTRIBUTE_KEY_PREFIX,
+            repositoryName
+        );
+        Map<String, String> settingsMap = node.getAttributes()
+            .keySet()
+            .stream()
+            .filter(key -> key.startsWith(settingsAttributeKeyPrefix))
+            .collect(Collectors.toMap(key -> key.replace(settingsAttributeKeyPrefix, ""), key -> validateAttributeNonNull(key)));
+
+        if (settingsMap.isEmpty()) {
+            throw new IllegalStateException(
+                "joining node [" + this.node + "] doesn't have settings attribute for [" + repositoryName + "] repository"
+            );
+        }
+
+        return settingsMap;
+    }
+
+    private RepositoryMetadata buildRepositoryMetadata(String name) {
+        String type = validateAttributeNonNull(String.format(Locale.getDefault(), REMOTE_STORE_REPOSITORY_TYPE_ATTRIBUTE_KEY_FORMAT, name));
+        Map<String, String> settingsMap = validateSettingsAttributesNonNull(name);
+
+        Settings.Builder settings = Settings.builder();
+        settingsMap.forEach(settings::put);
+
+        return new RepositoryMetadata(name, type, settings.build());
+    }
+
+    private RepositoriesMetadata buildRepositoriesMetadata() {
+        String segmentRepositoryName = validateAttributeNonNull(REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY);
+        String translogRepositoryName = validateAttributeNonNull(REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY);
+        if (segmentRepositoryName.equals(translogRepositoryName)) {
+            return new RepositoriesMetadata(Collections.singletonList(buildRepositoryMetadata(segmentRepositoryName)));
```

Comment:
this would become complex once we have more types of repos. Can we just find out unique repositories and add it to the list.